blog: kwctl sha support (#187)
kubewarden/kubewarden.io: content/blog/2023/09/kwctl-sha.md

---
title: kwctl SHA support
authors:
  - Fabrizio Sestito

> date: 2023-09-06

Reviewer: Reminder: change the date later, to match the day of the publication

Reviewer: fixed

---

> Lately, we have been focusing on improving the developer experience of Kubewarden, and in general,
> we are striving to implement features that are being requested by the community.

Reviewer 2: May I suggest, two simple sentences.:

```suggestion
Recently, we have focused on improving the Kubewarden developer experience.
We have been implementing features requested by the community.
```

# Reference policies by their SHA

> Today we are happy to announce that we have released a new version of `kwctl` that supports

Reviewer 3: I think we should change this sentence; the publish date of the post will not match the release of 1.7.0, and as it is worded, it feels we did a new release. I understand that the "today" is for only "being happy to announce" and not "we have released", but it's ambiguous.

Maybe something like:
> This effort got rewarded, and since the `v1.7.0` release `kwctl` now supports referencing policies by their SHA.

> Today we are happy to announce that we have released a new version of `kwctl` that supports
> referencing policies by their SHA.

Reviewer 2: ```suggestion
Since `kwctl` release `v1.7.0` we support referencing policies by their SHA.
```

Container engines such as Docker and Podman allow users to refer to images by their SHA sum.

> Being Kubewarden policies distributed as OCI artifacts, we tought it would be a good idea

Reviewer 3: ```suggestion
Being that Kubewarden policies are distributed as OCI artifacts, we thought it would be a good idea
```

> to add the SHA support to `kwctl`, so that the user could find a familiar experience.

Reviewer 3: ```suggestion
to add the SHA support to `kwctl`, so that the user has a familiar experience.
```

(usually, having less passive voices is easier for reading)

> Being Kubewarden policies distributed as OCI artifacts, we tought it would be a good idea
> to add the SHA support to `kwctl`, so that the user could find a familiar experience.

Reviewer 2: ```suggestion
As  Kubewarden policies are distributed as OCI artifacts, we thought it would be a good idea
to add the SHA support to `kwctl`, so that users have a familiar experience.
```

> Now, you can use a SHA prefix to refer to a policy which was previously pulled, and `kwctl` will automatically resolve its full SHA.

Reviewer 2: ```suggestion
Now, you can use a SHA prefix to refer to a policy which was previously pulled, and `kwctl` automatically resolves its full SHA.
```
